internal/util/helpers_test.go

```diff
@@ -4,18 +4,758 @@
+			fileContent: `{"access_token":"access_token","refresh_token":"refresh_token","expiry":"2024-10-05T17:46:27+10:00"}`,
+			expectedResult: util.OpenIdCredentials{
+				AccessToken:          "access_token",
+				RefreshToken:         "refresh_token",
+				AccessTokenExpiresAt: time.Date(2024, 10, 5, 17, 46, 27, 0, time.FixedZone("AEST", 10*60*60)),
+			},
+		},
+		{
+			name:          "Error unmarshaling credentials",
+			fileContent:   `invalid_json`,
+			expectedError: "error unmarshaling credentials: invalid character 'i'",
+		},
+		{
+			name:          "Error reading credentials file",
+			fileContent:   "",
+			expectedError: "error reading credentials file",
+		},
+	}
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tempDir := t.TempDir()
+			defer os.RemoveAll(tempDir)
+			// Create the minder directory inside the temp directory
+			minderDir := filepath.Join(tempDir, "minder")
+			err := os.MkdirAll(minderDir, 0750)
+			if err != nil {
+				t.Fatalf("failed to create minder directory: %v", err)
+			}
+
+			filePath := filepath.Join(minderDir, "credentials.json")
+
+			if tt.fileContent != "" {
+				// Create a temporary file with the specified content
+				err := os.WriteFile(filePath, []byte(tt.fileContent), 0600)
+				if err != nil {
+					t.Fatalf("failed to write test file: %v", err)
+				}
+				// Print the file path and content for debugging
+				t.Logf("Test %s: written file path %s with content: %s", tt.name, filePath, tt.fileContent)
+			} else {
+				// Print the file path for debugging
+				t.Logf("Test %s: file path %s not created as file content is empty", tt.name, filePath)
+			}
+
+			// Temporarily override the environment variable for the test
+			originalEnv := os.Getenv("XDG_CONFIG_HOME")
+			err = os.Setenv("XDG_CONFIG_HOME", tempDir)
+			if err != nil {
+				t.Errorf("error setting XDG_CONFIG_HOME: %v", err)
+			}
+			// reset this the environment variable when complete.
+			defer os.Setenv("XDG_CONFIG_HOME", originalEnv)
+
+			result, err := util.LoadCredentials()
+			if tt.expectedError != "" {
+				if err == nil || !strings.HasPrefix(err.Error(), tt.expectedError) {
+					t.Errorf("expected error matching %v, got %v", tt.expectedError, err)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("expected no error, got %v", err)
+				}
+				if result.AccessToken != tt.expectedResult.AccessToken || result.RefreshToken != tt.expectedResult.RefreshToken || !result.AccessTokenExpiresAt.Equal(tt.expectedResult.AccessTokenExpiresAt) {
+					t.Errorf("expected result %v, got %v", tt.expectedResult, result)
+				}
+			}
+
+		})
+
+	}
+}
+
+type TestCase struct {
+	name         string
+	token        string
+	issuerUrl    string
+	clientId     string
+	tokenHint    string
+	expectedPath string
+	expectError  bool
+	createServer func(t *testing.T, tt TestCase) *httptest.Server
+}
+
+func createTestServer(t *testing.T, tt TestCase) *httptest.Server {
```

Comment:
golangci-lint likes non-top-level test functions that take a `t *testing.T` to call `t.Helper()` so that the tracing of errors from `t.Errorf()` etc gets a cleaner stacktrace.

Reply:
t.Helper() is added to createTestServer().